modules/platform/nuxeo-drive-server/nuxeo-drive-mongodb/src/main/java/org/nuxeo/drive/mongodb/MongoDBAuditChangeFinder.java

```diff
@@ -0,0 +1,232 @@
+
+        if(queryStart == null) {
+            try (InputStream is = getClass().getClassLoader().getResourceAsStream("OSGI-INF/queryStart.json")) {
+                queryStart = IOUtils.toString(is, "UTF-8");
+            } catch (IOException e) {
+                log.error("couldn't read query start template", e);
+            }
+        }
+        queryBuilder.append(queryStart);
+        // first conditional filter
+        if (!activeRoots.getPaths().isEmpty()) {
+            queryBuilder.append(",\n");
+            // We open an $or array here in case the next conditional filter is needed as it's this or that
+            queryBuilder.append("{ \"$or\" : [");
+            queryBuilder.append(getCurrentRootFilteringClause(activeRoots.getPaths(), params) + "\n");
+        }
+        // second conditional filter
+        if (collectionSyncRootMemberIds != null && !collectionSyncRootMemberIds.isEmpty()) {
+            queryBuilder.append(",\n");
+            queryBuilder.append(getCollectionSyncRootFilteringClause(collectionSyncRootMemberIds, params) + "\n");
+        }
+        // if the first conditional filter was set, then an $or array has been opened and needs to be closed
+        if (!activeRoots.getPaths().isEmpty()) {
+            queryBuilder.append("]}\n");
+        }
+        if(queryEnd == null) {
+            try (InputStream is = getClass().getClassLoader().getResourceAsStream("OSGI-INF/queryEnd.json")) {
+                queryEnd = IOUtils.toString(is, "UTF-8");
+            } catch (IOException e) {
+                log.error("couldn't read query end template", e);
+            }
+        }
+        queryBuilder.append(queryEnd);
+
+        Bson filter = auditService.buildFilter(queryBuilder.toString(), params);
+        var sorts = List.of(Sorts.ascending("repositoryId"), Sorts.descending("eventDate"));
+        var order = Sorts.orderBy(sorts);
+        FindIterable<Document> iterable = auditCollection.find(filter).sort(order);
+        List<LogEntry> entries = StreamSupport.stream(iterable.spliterator(), false)
+                                              .map(MongoDBAuditEntryReader::read)
+                                              .collect(Collectors.toList());
+
+        // Post filter the output to remove (un)registration that are unrelated
+        // to the current user.
+        List<LogEntry> postFilteredEntries = new ArrayList<>();
+        String principalName = session.getPrincipal().getName();
+        for (LogEntry entry : entries) {
+            ExtendedInfo impactedUserInfo = entry.getExtendedInfos().get("impactedUserName");
+            if (impactedUserInfo != null && !principalName.equals(impactedUserInfo.getValue(String.class))) {
+                // ignore event that only impact other users
+                continue;
+            }
+            log.debug("Change detected: {}", entry);
+            postFilteredEntries.add(entry);
+        }
+        return postFilteredEntries;
```

Comment:
This could be expressed in the MongoDB query?

Reply:
Could be yes, that's not an option we had in JPA but here if the underlying structure of `impactedUserName` can be queried directly it'll be more efficient.

Reply:
To be continued

Reply:
It seems to me we never fall into continue here because the impactedUserName property does not exist.
None of the existing tests covers this post filtering aspect which seems to not be working to me.
I Added this to the mongo query (using the extended.user property in place of `impactedUserName`) and it seems test get broken when filtering really works.

Reply:
This was added by 6b8901e8954b2ec7ce9ffb4ac9035f3e7ba44acc but indeed I can't find any use

Reply:
How about removing this whole chunk of code since it has never been of use and since nobody complained about wrong behavior?

Reply:
Ok with me. @kevinleturc ?

Reply:
Ok too 👍

Reply:
Seems it's set here: https://github.com/nuxeo/nuxeo/blob/master/modules/platform/nuxeo-drive-server/nuxeo-drive-core/src/main/java/org/nuxeo/drive/listener/NuxeoDriveFileSystemDeletionListener.java#L249

Reply:
Indeed and it's set by https://github.com/nuxeo/nuxeo/blob/11ef171/modules/platform/nuxeo-drive-server/nuxeo-drive-core/src/main/java/org/nuxeo/drive/service/impl/NuxeoDriveManagerImpl.java#L313 and `fireEvent` is called with a specific userName.

So I retract my comment and we should keep this code.

Reply:
Instead of keeping the code. How about adding this filter at the end of the root `AND`
```
Bson isUser = and(or(exists("extended.impactedUser", true), eq("extended.user", session.getPrincipal().getName())));
```
Then a test covering that. I see some logs are generated by System and some are generated by Administrator so it should be easy to test.

Reply:
While working more on the subject I have noticed my debug queries are always printed twice. I think this is why:
https://github.com/nuxeo/nuxeo/blob/master/modules/platform/nuxeo-drive-server/nuxeo-drive-core/src/main/java/org/nuxeo/drive/service/impl/AuditChangeFinder.java#L101-L116

We query once the log entries and if there is a change in the drive roots, we update them and do the query again.

The check done after the first query could be done as a simple query rather than post filtering the log entries.
Wdyt ?

Reply:
https://github.com/nuxeo/nuxeo/blob/master/modules/platform/nuxeo-drive-server/nuxeo-drive-core/src/main/java/org/nuxeo/drive/service/impl/AuditChangeFinder.java#L69 queries the logs.
If it finds NuxeoDrive events it invalidates the roots cache and does the same query again:
https://github.com/nuxeo/nuxeo/blob/master/modules/platform/nuxeo-drive-server/nuxeo-drive-core/src/main/java/org/nuxeo/drive/service/impl/AuditChangeFinder.java#L101-L117

I have factorized things so invalidation stays unique but instead of doing that for loop I proposed making a mongo query. The change has been rejected because doing an extra query has been deemed too expensive while doing the same query twice just relies on costless cached results (efge will be able to explain more). However If you want to see, I have stashed it.